rpc-broker/src/msg.c

```diff
@@ -0,0 +1,245 @@
+/*
+ Copyright (c) 2018 AIS, Inc.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#include "rpc-broker.h"
+
+
+/*
+ * All requests are handled by this function whether they are raw requests
+ * coming from port 5555 or Websocket requests from 8080.  For every policy
+ * rule listed either in /etc/rpc-broker.policy file or the domain-specific
+ * rules listed in xenclient database are passed to `filter` to determine
+ * whether or not to be dropped or passed through.
+ */
+int broker(struct dbus_message *dmsg, int domid)
+{
+    if (!dmsg) {
+        DBUS_BROKER_WARNING("Invalid args to broker-request %s", "");
+        return 1;
+    }
+
+    if (!dbus_broker_policy) {
+        DBUS_BROKER_WARNING("No policy in place %s", "");
+        return 1;
+    }
+
+    int policy = 0;
+
+    struct etc_policy etc = dbus_broker_policy->etc;
+
+    int i, j;
+    for (i=0; i < etc.count; i++)
+        policy = filter(&(etc.rules[i]), dmsg, domid);
```

Comment:
This enforce last rule that matched. Does that align with the current behavior of rpc-proxy?

Reply:
Yes, the [rules format](https://openxt.atlassian.net/wiki/spaces/DC/pages/15368227/Toolstack+Guide#ToolstackGuide-RPCrulesformat) is structured as last match.  By default "deny all" is the first rule and after any following rules can be applied to override this.  Do you have any thoughts or opinions on the way this is currently setup?